src/lib.rs

```diff
@@ -1172,6 +1170,12 @@ pub mod pallet {
 			(session_total_amount, unclaimable_rewards)
 		}
 
+		fn has_duplicates(accounts: &[T::AccountId]) -> bool {
+			let duplicates =
+				accounts.iter().collect::<sp_std::collections::btree_set::BTreeSet<_>>();
```

Comment:
Shouldn't this receive `accounts:  &BoundedVec<T::AccountId, MaxInvulnerables>` to avoid unbounded loop iterations? I understand that the invulnerables are going to be a small set of accounts, but as a good practice we should keep the iteration bounded.

Reply:
This function in also used in the genesis configuration, where there is no weight limit. We could indeed change parameter in `set_invulnerables` to make sure it is a bounded vector, but still in this function we are passing a slice that is meant to be controlled by the caller. Meaning that this function should have no notion nor control whether the bounds are actually respected, and IMO it should be controlled outside.

Reply:
In that case, just to keep the bounds you could automatically create the boundedVec inside the function and throw an error if the size of the received array is incorrect. E.g.:
```rust
fn has_duplicates(account &[T::AccountId]) -> bool {
  let bounded_accounts: BoundedVec<T::AccountId, MaxInvulnerables> =
				accounts.try_into().map_err(|_| Error::<T>::TooManyInvulnerables)?;
  //...
}

Reply:
It would suffice to check the slice length against the maximum one.

In any case the function would not return a `bool`, but a `Result`, which is not ideal. I don't think it's such a hassle to check outside the function, as in any case it is a private one that can only be used within the pallet's context.